### ISSUE TITLE
Impossible to configure minor/major grid line style independently in rcParams

### ISSUE THREAD

**Issue by NGJROMO** [matplotlib/matplotlib#13919]
Hi,

As far as I can see, the only customization to the grid that is possible right now in rcParams is:

```
mpl.rcParams['grid.color']  
mpl.rcParams['grid.alpha']
mpl.rcParams['grid.linewidth']
mpl.rcParams['grid.linestyle']
```

Is it possible to change these keys so that it is possible to configure minor and major grid line styles independently from each other?

Something like:

```
mpl.rcParams['grid.minor.color']  
mpl.rcParams['grid.minor.alpha']
mpl.rcParams['grid.minor.linewidth']
mpl.rcParams['grid.minor.linestyle']
mpl.rcParams['grid.major.color']  
mpl.rcParams['grid.major.alpha']
mpl.rcParams['grid.major.linewidth']
mpl.rcParams['grid.major.linestyle']
```

**Comment by timhoffm** [comment]
There are no separate parameters for major/minor grid properties in `rcParams`.

You can currently only configure them programmatically.

~~~
import matplotlib.pyplot as plt
from matplotlib.ticker import MultipleLocator

x = np.arange(10)

ax = plt.gca()
ax.plot(x, x**2)
ax.xaxis.set_minor_locator(MultipleLocator(0.5))
ax.grid(which='both')
ax.tick_params(which='minor', grid_linestyle='--')
~~~

**Comment by zertrin** [comment]
I was just looking for exactly this feature. I already did it programmatically before, but as I have a large number of different plots with major and minor gridlines, this is quite inconvenient.

Looking forward such an option in rcParams!

**Comment by dgua** [comment]
Yes, **PLEASE**. I'm with @NGJROMO. 
The programmatic way shown above is too cumbersome (and, is there a way to also set line width, color, and alpha?). 

This is back to one year ago. Is there any way it can be implemented soon?

Here is what I want:
![ExamplePlot-Two Grids](https://user-images.githubusercontent.com/3887028/78744224-e2871880-7915-11ea-94d7-2ff336257e62.png)

Thanks.

**Comment by hjstein** [comment]
I'm surprised that the settings are not automatically mirrored in the style sheet & rcParams.  In fact, I would have thought that all of the defaults would be set up in rcParams & that would drive everything else.

**Comment by neutrinonerd3333** [comment]
Bumping this issue -- I've found it unwieldy to copy and paste the same code snippet to get a fixed style that I almost always want, and would also appreciate support for differently styled major and minor gridlines via rc params.

**Comment by jklymak** [comment]
A motivated user could always write a PR for this.  https://matplotlib.org/stable/devel/contributing.html  However, be aware that new rcparams face a relatively high bar for acceptance.

One can also write their own styling wrapper if they are always doing this.

```python
import matplotlib.pyplot as plt 

my_style_subplots(*arg, **kwargs):
    fig, axs = plt.subplots(*args, **kwargs)
    for ax in axs.flat:
       ax.xaxis.set_minor_locator(MultipleLocator(0.5))
       ax.grid(which='both')
       ax.tick_params(which='minor', grid_linestyle='--')
    return fig, axs
```

```python
import mystyles as myplt

fig, ax = mplt.subplots(3, 2, sharedx=True)
```

**Comment by dgua** [comment]
Sorry but above answer isn't satisfactory IMHO. That is not what we are asking. Also, it forces ax.xaxis.set_minor_locator(MultipleLocator(0.5)), which I would not want in general. 

This is a straightforward request and not hard to implement. It's from 2019 for crying out loud.

**Comment by jklymak** [comment]
> This is a straightforward request and not hard to implement. It's from 2019 for crying out loud.

Matplotlib is a volunteer organization and PRs are always welcome!

**Comment by DanHallatt** [comment]
I agree with @neutrinonerd3333, and am in the same boat as him and @zertrin.

One other thing that I think is related, is the inability to specify tick label font in rcParams. I am able to specify the overall general font as Helvetica for example, successfully changing the font of everything in my generated plots, however this does not change the font of the tick labels (they remain in a serif font..)! I thought I could specify this in the rcParams, but it turns out this is another limitation that must be done programmatically... making each of my plots even more cumbersome and repetitive to code.

**Comment by timhoffm** [comment]
This is a highly requested feature. -> keep

**Comment by YamenMejri** [comment]
any updates on this?

**Comment by nNeD5** [comment]
Still need this feature

**Comment by jklymak** [comment]
Maybe some extra tagging will help, but someone who wants this feature needs to propose a PR for it to get implemented.

**Comment by timhoffm** [comment]
As a word of warning: This is not as trivial as it sounds. We want to maintain backward-compatibility, i.e. if somebody changes th `grid.*` parameters, they should still affect both grids - unless of course they specify `grid.minor.*` explicitly. This possibly needs some hierachial lookup/dependency between the keys, which is not a concept we have in `rcParams` so far.

**Comment by carlosgmartin** [comment]
@QuLogic @timhoffm The new options, like `grid.major.color`, don't show up in ["The default `matplotlibrc` file"](https://matplotlib.org/stable/users/explain/customizing.html#the-default-matplotlibrc-file) in the docs. Is this expected?

**Comment by timhoffm** [comment]
@carlosgmartin As you can see from the milestone assignment, this will be released in 3.11. The stable docs are currently 3.10.7. You can find the dev version of the docs at https://matplotlib.org/devdocs/users/explain/customizing.html
